### ISSUE TITLE
ContinueAsNew to a different workflow fails

### ISSUE THREAD

**Issue by johncburns1** [nforgeio/neonKUBE#667]
The ContinueAsNew tests that continue with different workflow stubs never stop continuing.

**Comment by jefflill** [comment]
These unit tests are failing:
```
Test.EndToEnd.Workflow_ContinueAsNew_Stub
Test.EndToEnd.Workflow_ContinueAsNew_StubOptions
```
These tests are currently disabled.

**Comment by johncburns1** [comment]
I'll debug and try to resolve.

**Comment by johncburns1** [comment]
So I resolved thew issue with `Test.EndToEnd.Workflow_ContinueAsNew_StubOptions`.  I added a `Workflow` field to `ContinueAsNewOptions`, which allows a user to specify the string name of a workflow they want to execute on restart.  On the `cadence-proxy` side, this allows us to create the `ContinueAsNewError` with either the original workflow name a specified one if provided. This is a little more difficult with `Test.EndToEnd.Workflow_ContinueAsNew_Stub` because without passing the options (unless we want to pass a `string workflow` param) we have no way of knowing about the new workflow we want to execute upon restart.  So `ContinueAsNew` with a different workflow might only work if `ContinueAsNewOptions` are provided.
